src/msrc_file_system_depth_device.cpp

```diff
@@ -0,0 +1,69 @@
+#include <boost/filesystem.hpp>
+#include <opencv2/opencv.hpp>
+#include <seng499/msrc_file_system_depth_device.hpp>
+#include <regex>
+#include <iostream>
+#include <cstdint>
+
+namespace seng499 {
+
+	static const std::regex & get_regex() {
+
+		static const std::regex msrc_depth_file_pattern(".*depth.*");
+		return msrc_depth_file_pattern;
+
+	}
+	
+	
+	bool msrc_file_system_depth_device_filter::operator () (const boost::filesystem::path & path) const {
+		
+		return std::regex_match(path.filename().string(), get_regex());
+		
+	}
+	
+	
+	std::vector<float> msrc_file_system_depth_device_frame_factory::operator () (const boost::filesystem::path & path, std::vector<float> vec) {
+	
+		// make sure this is empty
+		vec.clear();
+
+		// reserve the size of our image
+		vec.reserve(640*480);
+
+		// Load a grayscale depth image. Anydepth is necessary as these are 16 bit ints 
+		cv::Mat img = cv::imread(path.native(), CV_LOAD_IMAGE_GRAYSCALE | CV_LOAD_IMAGE_ANYDEPTH);
```

Comment:
Should we verify that the file read in is actually 640x480 and throw if it's not?